File: typescript/1.两数之和.js
/*
 * @lc app=leetcode.cn id=1 lang=javascript
 *
 * [1] 两数之和
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let m = new Map()
    for (let i = 0; i < nums.length; i++) {
        if (m.has(target - nums[i]))
            return [m.get(target - nums[i]), i]
        m.set(nums[i], i)
    }
    return []
};
// @lc code=end

File: typescript/1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        for i in range(len(nums)):
            if m.get(target-nums[i]) is not None:
                return [m.get(target-nums[i]), i]
            m[nums[i]] = i
        return []
# @lc code=end
